node/index.ts

```diff
@@ -8,14 +8,64 @@ import * as SignalClient from './libsignal_client';
 
 const SC = bindings('libsignal_client') as typeof SignalClient;
 
+export class PublicKey {
+  private readonly nativeHandle: SignalClient.PublicKey;
+
+  private constructor(handle: SignalClient.PublicKey) {
+    this.nativeHandle = handle;
+  }
+
+  static from_native_handle(handle: SignalClient.PublicKey): PublicKey {
+    return new PublicKey(handle);
+  }
+
+  static deserialize(buf: Buffer): PublicKey {
+    return new PublicKey(SC.PublicKey_deserialize(buf));
+  }
+
+  serialize(): Buffer {
+    return SC.PublicKey_serialize(this.nativeHandle);
+  }
+
+  verify(msg: Buffer, sig: Buffer): boolean {
+    return SC.PublicKey_verify(this.nativeHandle, msg, sig);
+  }
+
+  getNativeHandle(): SignalClient.PublicKey {
```

Comment:
Should we add tests for this?

Reply:
There isn't anything that JavaScript world can meaningfully do with the result. The only reason it is not private is that operations that need to take in a JS PublicKey and pass it to a Rust operation needs to be able to get at the underlying handle (see for example in PrivateKey.agree). Ideally this would be package private; nothing outside of this file should ever call it. But I couldn't find any way of doing that.

Maybe, given that this will be accessible to anyone using the library, it should be given a scarier name like `_unsafeGetNativeHandle`.

Reply:
I'd prefer that name—makes it clearer that it's only public because of a weakness of JavaScript.

Reply:
Done